From a1cc4c9efacc846ba837e4f03bee89d4ac0f4a1d Mon Sep 17 00:00:00 2001
Message-ID: <a1cc4c9efacc846ba837e4f03bee89d4ac0f4a1d.1707646257.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1707646257.git.sharlatanus@gmail.com>
References: <cover.1707646257.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 10 Feb 2024 17:00:55 +0000
Subject: [PATCH 19/30] gnu: go-golang.org-x-sync-errgroup: Move to (gnu
 packages-build).

* gnu/packages/golang.scm (go-golang.org-x-sync-errgroup,
go-golang.org-x-sync-semaphore): Move from here ...
* gnu/packages/golang-build.scm: ... to here.

* gnu/packages/check.scm: Add (gnu packages golang-build) module.

Change-Id: Ie886c0f44e0a70629e23371b447f5a35006ade21
---
 gnu/packages/check.scm        |  1 +
 gnu/packages/golang-build.scm | 41 +++++++++++++++++++++++++++++++++++
 gnu/packages/golang.scm       | 38 --------------------------------
 gnu/packages/terminals.scm    |  1 +
 4 files changed, 43 insertions(+), 38 deletions(-)

diff --git a/gnu/packages/check.scm b/gnu/packages/check.scm
index 4f593cde8d..b520d6c58a 100644
--- a/gnu/packages/check.scm
+++ b/gnu/packages/check.scm
@@ -77,6 +77,7 @@ (define-module (gnu packages check)
   #:use-module (gnu packages glib)
   #:use-module (gnu packages gnome)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-build)
   #:use-module (gnu packages gtk)
   #:use-module (gnu packages guile)
   #:use-module (gnu packages guile-xyz)
diff --git a/gnu/packages/golang-build.scm b/gnu/packages/golang-build.scm
index fed606e8fb..8e64571929 100644
--- a/gnu/packages/golang-build.scm
+++ b/gnu/packages/golang-build.scm
@@ -1,11 +1,14 @@
 ;;; GNU Guix --- Functional package management for GNU
+;;; Copyright © 2020 Danny Milosavljevic <dannym@scratchpost.org>
 ;;; Copyright © 2020 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2020 Ryan Prior <rprior@protonmail.com>
+;;; Copyright © 2020 Vagrant Cascadian <vagrant@debian.org>
 ;;; Copyright © 2021 Ludovic Courtès <ludo@gnu.org>
 ;;; Copyright © 2021 hackeryarn <artemchernyak@gmail.com>
 ;;; Copyright © 2023 Artyom V. Poptsov <poptsov.artyom@gmail.com>
 ;;; Copyright © 2023 Felix Lechner <felix.lechner@lease-up.com>
 ;;; Copyright © 2023 Katherine Cox-Buday <cox.katherine.e@gmail.com>
+;;; Copyright © 2023 Timo Wilken <guix@twilken.net>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -165,6 +168,44 @@ (define-public go-golang-org-x-net-html
     (description
      "This package provides an HTML5-compliant tokenizer and parser.")))
 
+(define-public go-golang.org-x-sync-errgroup
+  (let ((commit "cd5d95a43a6e21273425c7ae415d3df9ea832eeb")
+        (revision "0"))
+    (package
+      (name "go-golang.org-x-sync-errgroup")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://go.googlesource.com/sync")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "1nqkyz2y1qvqcma52ijh02s8aiqmkfb95j08f6zcjhbga3ds6hds"))))
+      (build-system go-build-system)
+      (arguments
+       '(#:import-path "golang.org/x/sync/errgroup"
+         #:unpack-path "golang.org/x/sync"))
+      (home-page "https://godoc.org/golang.org/x/sync/errgroup")
+      (synopsis "Synchronization, error propagation, and Context cancellation
+for groups of goroutines working on subtasks of a common task")
+      (description "This package provides synchronization, error
+propagation, and Context cancellation for groups of goroutines working on
+subtasks of a common task.")
+      (license license:bsd-3))))
+
+(define-public go-golang.org-x-sync-semaphore
+  (package
+    (inherit go-golang.org-x-sync-errgroup)
+    (name "go-golang.org-x-sync-semaphore")
+    (arguments
+     '(#:import-path "golang.org/x/sync/semaphore"
+       #:unpack-path "golang.org/x/sync"))
+    (home-page "https://godoc.org/golang.org/x/sync/semaphore")
+    (synopsis "Weighted semaphore implementation in Go")
+    (description "Weighted semaphore implementation in Go.")))
+
 (define-public go-golang-org-x-sys
   (let ((commit "ca59edaa5a761e1d0ea91d6c07b063f85ef24f78")
         (revision "0"))
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 68b905ff8b..7e5626a303 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -7023,44 +7023,6 @@ (define-public go-github-com-google-renameio
 create or replace a file or symbolic link.")
     (license license:asl2.0)))
 
-(define-public go-golang.org-x-sync-errgroup
-  (let ((commit "cd5d95a43a6e21273425c7ae415d3df9ea832eeb")
-        (revision "0"))
-    (package
-      (name "go-golang.org-x-sync-errgroup")
-      (version (git-version "0.0.0" revision commit))
-      (source (origin
-                (method git-fetch)
-                (uri (git-reference
-                       (url "https://go.googlesource.com/sync")
-                       (commit commit)))
-                (file-name (git-file-name name version))
-                (sha256
-                 (base32
-                  "1nqkyz2y1qvqcma52ijh02s8aiqmkfb95j08f6zcjhbga3ds6hds"))))
-      (build-system go-build-system)
-      (arguments
-       '(#:import-path "golang.org/x/sync/errgroup"
-         #:unpack-path "golang.org/x/sync"))
-      (synopsis "Synchronization, error propagation, and Context cancellation
-for groups of goroutines working on subtasks of a common task")
-      (description "This package provides synchronization, error propagation,
-and Context cancellation for groups of goroutines working on subtasks of a
-common task.")
-      (home-page "https://godoc.org/golang.org/x/sync/errgroup")
-      (license license:bsd-3))))
-
-(define-public go-golang.org-x-sync-semaphore
-  (package
-    (inherit go-golang.org-x-sync-errgroup)
-    (name "go-golang.org-x-sync-semaphore")
-    (arguments
-     '(#:import-path "golang.org/x/sync/semaphore"
-       #:unpack-path "golang.org/x/sync"))
-    (synopsis "Weighted semaphore implementation in Go")
-    (description "Weighted semaphore implementation in Go.")
-    (home-page "https://godoc.org/golang.org/x/sync/semaphore")))
-
 (define (go-gotest-tools-source version sha256-base32-hash)
   (origin
     (method git-fetch)
diff --git a/gnu/packages/terminals.scm b/gnu/packages/terminals.scm
index f609de8eff..37223d0d18 100644
--- a/gnu/packages/terminals.scm
+++ b/gnu/packages/terminals.scm
@@ -90,6 +90,7 @@ (define-module (gnu packages terminals)
   #:use-module (gnu packages glib)
   #:use-module (gnu packages gnome)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-build)
   #:use-module (gnu packages gtk)
   #:use-module (gnu packages image)
   #:use-module (gnu packages libcanberra)
-- 
2.41.0

